src/memray/_memray/hooks.cpp

```diff
@@ -1,5 +1,7 @@
 #include <cassert>
 #include <cstdio>
+#include <mutex>
+#include <unordered_set>
```

Comment:
Hm - you did `#include <unordered_set>` but used `std::set` in the implementation.

Either you included the wrong thing or you used the wrong thing, I think.